sdk/storage/azure-storage-blob/tests/test_blob_client.py

```diff
@@ -731,4 +733,25 @@ def test_create_configuration_legacy(self, **kwargs):
         assert config.max_block_size == 4 * 1024 * 1024
         assert sdk_name in config.user_agent_policy.user_agent
 
+    @BlobPreparer()
+    @recorded_by_proxy
+    def test_bad_storage_account_key_no_retry(self, **kwargs):
+        storage_account_name = kwargs.pop("storage_account_name")
+        storage_account_key = "a"
+
+        blob_client = BlobClient(
+            account_url=self.account_url(storage_account_name, "blob"),
+            container_name="foo",
+            blob_name="bar",
+            credential=storage_account_key,
+        )
+
+        start = time.time()
+        with pytest.raises(AzureSigningError) as e:
+            blob_client.get_blob_properties()
```

Comment:
A better way to test this would be with the `retry_hook` keyword. You can pass a function to any method call using `retry_hook=func` and in that function you could increment a retry count or something. Additionally, I don't think this test needs to be recorded as it doesn't actually make any requests (it errors out before making one). So, you can remove `recorded_by_proxy` and revert the assets file.

Maybe also move this test to `test_retry.py` instead? Feels better suited for this.